File: backend/src/main/java/ink/whi/backend/dao/entity/KnowledgeBase.java
package ink.whi.backend.dao.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import ink.whi.backend.common.enums.ProcessTypeEnum;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author: qing
 * @Date: 2025/6/18
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("knowledge_base")
public class KnowledgeBase extends BaseEntity {

    /**
     * 标题
     */
    private String title;

    /**
     * 描述
     */
    private String remark;

    private Boolean isPublic;

    private Integer ownerId;

    /**
     * 嵌入模型id
     */
    private Long embeddingModelId;

    /**
     * 问答模型
     */
    private Integer qaModelId;

    // =================数据处理设置==================

    /**
     * 处理方式
     */
    private ProcessTypeEnum processType;

    /**
     * 分块大小(100-3000)
     */
    private Integer blockSize;

    /**
     * 文档切割时重叠数量(按token来计)
     */
    private Integer maxOverlap;

    /**
     * 问答提示词（可选）
     */
    private String qaPrompt;


    // ==================召回设置==================

    /**
     * 文档召回最大数量
     */
    private Integer retrieveMaxResults;

    /**
     * 文档召回最小分数
     */
    private Double retrieveMinScore;
}

File: web/src/types/knowledgeBase.ts
// Knowledge Base types
export interface KnowledgeBase {
  id?: number;
  createTime?: string;
  updateTime?: string;
  title: string;
  remark?: string;
  isPublic?: boolean;
  ownerId?: number;
  embeddingModelId?: number;
  qaModelId?: number;
  processType?: string;
  blockSize?: number;
  maxOverlap?: string;
  qaPrompt?: string;
  retrieveMaxResults?: number;
  retrieveMinScore?: number;
}

export interface KbItemDto {
  id?: number;
  createTime?: string;
  updateTime?: string;
  title?: string;
  processType?: string;
  chunks?: number;
  embeddingStatus?: string;
  isEnable?: boolean;
}

export interface KbDetailVO {
  knowledgeBase: KnowledgeBase;
  itemLists?: KbItemDto[];
}

export interface SimpleKbDto {
  id?: number;
  createTime?: string;
  updateTime?: string;
  title?: string;
  remark?: string;
  isPublic?: boolean;
}